kubectl: support regional GKE clusters

mickybart opened the pull request:
Permit to use the region for regional clusters (zonal clusters are still supported)

new env: CLOUDSDK_COMPUTE_REGION

mickybart commented:
I signed it!

mickybart commented:
I signed it!

ImJasonH commented:
Fixes #176

cudobuild commented on kubectl/kubectl.bash:
```diff
@@ -2,24 +2,34 @@
+      echo "Running: gcloud config set container/use_v1_api_client false"
+      gcloud config set container/use_v1_api_client false
+      echo "Running: gcloud beta container clusters get-credentials --project=\"$project\" --region=\"$region\" \"$cluster\""
+      gcloud beta container clusters get-credentials --project="$project" --region="$region" "$cluster" || exit
```
Looks like this is GA now:

```suggestion
      gcloud container clusters get-credentials --project="$project" --region="$region" "$cluster" || exit
```

namusyaka commented:
Any progress on this? As @cudobuild said, I think you can fix it pointed by him.

mickybart commented:
Hi @suzutatsu , I merged your PR on my github but that closed this PR. I reopened it but as there is a coauthor (you), it seems not able to pass the Contributor License Agreement for you.
Can you please check on your side ? (https://cla.developers.google.com/)
Thanks

ImJasonH commented:
I was hoping to get some feedback from @DazWilkin but otherwise it looks reasonable to me.

DazWilkin commented:
Apologies: LGTM.

suzutatsu commented:
I signed it.

suzutatsu commented:
Hi @mickybart , Thank you for merge.

suzutatsu commented:
I signed it!

mickybart commented:
thanks @suzutatsu.

Comment to inform the bot as requested

In order to pass this check, please resolve this problem and have the pull request author add another comment and the bot will run again. If the bot doesn't comment, it means it doesn't think anything has changed.

suzutatsu commented:
I found a bug. Please wait a little.

suzutatsu commented:
I Fixed it.

ImJasonH commented:
@mickybart You still need to sign the CLA, @suzutatsu has already signed it.

ImJasonH commented:
#456 was just merged which conflicts with this change, sorry about that. Looks like a straightforward merge though.

mickybart commented:
I rebased the PR to have a clean history on top of previous changes on #456
Docker build was broken by #454 in my side (kubectl package not found). I added a fix in this PR.

I signed it!

mickybart commented:
I signed CLA!

suzutatsu commented:
@mickybart #454 was reverted. (https://github.com/GoogleCloudPlatform/cloud-builders/pull/459)

mickybart commented:
ok thanks I will rebase.

mickybart commented:
Rebase done.
Basically I just removed the commit to fix Dockerfile due to #454 reverted.

I still don't get why cla/google is complaining :/

suzutatsu commented:
@ImJasonH What should I do?
